rasa/shared/core/domain.py

```diff
@@ -298,9 +298,12 @@ def merge_lists_of_dicts(
         if override or self.session_config == SessionConfig.default():
             combined[SESSION_CONFIG_KEY] = domain_dict[SESSION_CONFIG_KEY]
 
-        combined[KEY_INTENTS] = merge_lists_of_dicts(
-            combined[KEY_INTENTS], domain_dict[KEY_INTENTS], override
-        )
+        duplicates: Dict[Text, List[Text]] = {}
```

Comment:
Because you removed a method and introduced the duplicate extraction, I'd also add a brief comment about the steps to clarify why these are done.